### DISCUSSION TITLE
2.8 TFT touch problem

### DISCUSSION THREAD

**Discussion by alphalneo** [Bodmer/TFT_eSPI#3034] [Q&A - Touch]
Hi
I bought all in one TFT 2.8 with ESP32, I am able to use the display but not the touch using the TFT_eSPI
<img width="246" alt="Screenshot 2023-12-12 at 1 34 02 PM" src="https://github.com/Bodmer/TFT_eSPI/assets/147276073/27e1f3b8-3fcc-4686-ad68-7a3ab3508719">
<img width="513" alt="Screenshot 2023-12-12 at 1 33 57 PM" src="https://github.com/Bodmer/TFT_eSPI/assets/147276073/274190c8-31df-43e3-a008-fc08946f0ec2">

I configured like this
```
#define TFT_MISO 12
#define TFT_MOSI 13 // In some display driver board, it might be written as "SDA" and so on.
#define TFT_SCLK 14
#define TFT_CS   15  // Chip select control pin
#define TFT_DC   2  // Data Command control pin
#define TFT_RST  -1  // Reset pin (could connect to Arduino RESET pin)
#define TFT_BL   21  // LED back-light

#define TOUCH_CS 33     // Chip select pin (T_CS) of touch screen
```

However, the touch clock, for example, is using a different pin

I tried with TFT_TOUCH with this setting
```
#define DOUT 39  /* Data out pin (T_DO) of touch screen */
#define DIN  32  /* Data in pin (T_DIN) of touch screen */
#define DCS  33  /* Chip select pin (T_CS) of touch screen */
#define DCLK 25  /* Clock pin (T_CLK) of touch screen */
```
And the touch is working

Can I use the TFT and Touch with TFT_eSPI? and the most important how?

**Comment by larrybud2004** [comment]
You say the "touch is working". So what is the problem?

**Comment by alphalneo** [comment]
The touch doesn't work with the tft_eSPI I added tft_touch in order to have it working
Therefore, the hardware for the touch is working, but I am not able to use it with tft_espi

**Comment by ballistRC** [comment]
Check that TOUCH_CS is really being seen.  I not sure of your Tool slow but with Platformio, I had to set it in User_Setup.h in the library with my recent tests.

**Comment by Bodmer** [comment]
The touch controller must be on the same SPI bus as the display but with a separate chip select.  Note also that you may need to disconned the TFT_SDO pin on some displays,[ see here](https://github.com/Bodmer/TFT_eSPI/discussions/898).

**Comment by alphalneo** [comment]
Thank you I will try to modify the circuit

**Reply by smoochiee** [reply]
Have you modified your circuit?im having same problem

**Reply by alphalneo** [reply]
Yes I did, I cut the touch screen connection and wired to the display
And now is working
